File: system/ulib/zx/vmo.cpp
// Copyright 2016 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <lib/zx/vmo.h>

#include <zircon/syscalls.h>

namespace zx {

zx_status_t vmo::create(uint64_t size, uint32_t options, vmo* result) {
    return zx_vmo_create(size, options, result->reset_and_get_address());
}

} // namespace zx

File: garnet/public/rust/fuchsia-vfs/pseudo-fs-mt/src/file/vmo.rs
// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

//! File nodes backed by a VMO.  These are useful for cases when individual read/write operation
//! actions need to be visible across all the connections to the same file.

pub mod asynchronous;

/// Asynchronous is the default and, currently, the only implementation provided for VMO backed
/// pseudo files.
pub use asynchronous::{read_only, read_write};

pub(self) mod connection;
